src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -216,11 +242,32 @@ def __lt__(self, other):  # pylint: disable=too-many-return-statements
         return False
 
 
+def _cross_api_format_lt(api_version, other):
+    """LT strategy that supports if types are different.
+
+    For now, let's assume that any Semver is higher than any DateAPI
+    This fits KeyVault, if later we have a counter-example we'll update
+    """
+    def is_semver(version):
+        # Keep it stupid and simple
+        return "." in version
```

Comment:
Version: `1` is supposed to be a semver.

Reply:
Right, would you prefer `'-' not in version`?

Reply:
If KeyVault goes with semver, `-` is not a good indicator either.
https://semver.org

Reply:
I hate to say this, but maybe a regular express to match the string. If it follows format of `YYYY-mm-dd*` it is not a semver.

Reply:
I do not like the more explicit option. Why not simply attempt to parse it as a date and if it fails, assume it is semver?

Reply:
Yep, I can do that :). Very Pythonic indeed